Python 3 fixes

rodrigc opened the pull request.

rodrigc committed:
Use threading module instead of thread module

In Python 3, the module "thread" was renamed to "_thread",
and deprecated.  Use of module "threading" is preferred.

See:
https://www.python.org/dev/peps/pep-3108/#obsolete
https://www.python.org/dev/peps/pep-3108/#thread-deprecation

pypingou commented on kerneltest/app.py:
```diff
@@ -171,7 +171,7 @@ def is_admin(user):
 
     admins = APP.config['ADMIN_GROUP']
-    if isinstance(admins, basestring):
+    if isinstance(admins, str):
```
This will break the behavior for py2, so :-1: as is.

Using six we may be able to be both py2 and py3 compatible.

rodrigc committed:
Add Python six library to install requirements

rodrigc committed:
Use six.string_types instead of basestring

basestring was removed in Python 3.
See: https://www.python.org/dev/peps/pep-3137/#the-basestring-type

six.string_types will use basestring on Python 2, and str on Python 3
